src/bundle/Templating/Twig/Functions/UserTracking.php

```diff
@@ -8,83 +8,107 @@
 
 namespace EzSystems\EzRecommendationClientBundle\Templating\Twig\Functions;
 
+use eZ\Publish\API\Repository\Values\Content\Content;
 use eZ\Publish\Core\MVC\ConfigResolverInterface;
 use eZ\Publish\Core\MVC\Symfony\Locale\LocaleConverterInterface;
 use EzSystems\EzRecommendationClient\Helper\ContentTypeHelper;
 use EzSystems\EzRecommendationClient\Service\UserServiceInterface;
 use EzSystems\EzRecommendationClient\Value\Parameters;
+use Ibexa\Personalization\Config\Repository\RepositoryConfigResolverInterface;
 use Symfony\Component\HttpFoundation\RequestStack;
 use Twig\Environment as TwigEnvironment;
 use Twig\Extension\RuntimeExtensionInterface;
 
-final class UserTracking extends AbstractFunction implements RuntimeExtensionInterface
+final class UserTracking implements RuntimeExtensionInterface
 {
     /** @var \eZ\Publish\Core\MVC\ConfigResolverInterface */
     private $configResolver;
 
+    /** @var \EzSystems\EzRecommendationClient\Helper\ContentTypeHelper */
+    private $contentTypeHelper;
+
     /** @var \eZ\Publish\Core\MVC\Symfony\Locale\LocaleConverterInterface */
     private $localeConverter;
 
-    /** @var \EzSystems\EzRecommendationClient\Service\UserServiceInterface */
-    private $userService;
-
-    /** @var \EzSystems\EzRecommendationClient\Helper\ContentTypeHelper */
-    private $contentTypeHelper;
+    /** @var \Ibexa\Personalization\Config\Repository\RepositoryConfigResolverInterface */
+    private $repositoryConfigResolver;
 
     /** @var \Symfony\Component\HttpFoundation\RequestStack */
     private $requestStack;
 
+    /** @var \Twig\Environment */
+    private $twig;
+
+    /** @var \EzSystems\EzRecommendationClient\Service\UserServiceInterface */
+    private $userService;
+
     public function __construct(
         ConfigResolverInterface $configResolver,
-        LocaleConverterInterface $localeConverter,
-        UserServiceInterface $userService,
         ContentTypeHelper $contentTypeHelper,
+        LocaleConverterInterface $localeConverter,
+        RepositoryConfigResolverInterface $repositoryConfigResolver,
         RequestStack $requestStack,
-        TwigEnvironment $twig
+        TwigEnvironment $twig,
+        UserServiceInterface $userService
     ) {
-        parent::__construct($twig);
-
         $this->configResolver = $configResolver;
         $this->contentTypeHelper = $contentTypeHelper;
         $this->localeConverter = $localeConverter;
+        $this->repositoryConfigResolver = $repositoryConfigResolver;
         $this->requestStack = $requestStack;
+        $this->twig = $twig;
         $this->userService = $userService;
     }
 
     /**
-     * Renders simple tracking snippet code.
+     * @throws \Exception
+     */
+    public function trackUser(Content $content): ?string
```

Comment:
If we want to maintain BC compatibility, we might not enforce `Content` as type here, instead allowing `Content|int` for 3.3 versions, and triggering deprecation warning if `int` is used.

That is only relevant if we have any customers that actually use this feature.